torch/nn/parallel/distributed.py

```diff
@@ -864,6 +866,49 @@ def __init__(
 
         self._lazy_init_ran = False
 
+        # Register the AccumulateGrad post hooks even if DDP is not compiled. This
+        # can avoid compiling the hooks twice. The Python hooks will be
```

Comment:
why does adding the hooks avoid compiling hooks twice?

Reply:
If we dynamically detect if `compiled_autograd()` is enabled in the backward, and register the hook, it seems to trigger the compilation twice. Let me remove this comment since this cause more confusion than what it tries to explain.